src/Blazored.Video/VideoQueue.cs

```diff
@@ -0,0 +1,255 @@
+		/// </summary>
+		/// <returns></returns>
+		public ValueTask PlayNext()
+		{
+			return TryPlayItem(VideoItems
+				.SkipWhile(e => e.Id != CurrentItem.Id)
+				.Skip(1)
+				.FirstOrDefault());
+		}
+
+		/// <summary>
+		///		Skips the current item and plays the previous in queue.
+		/// </summary>
+		/// <returns></returns>
+		public ValueTask PlayPrevious()
+		{
+			return TryPlayItem(VideoItems
+				.Reverse()
+				.SkipWhile(e => e.Id != CurrentItem.Id)
+				.Skip(1)
+				.FirstOrDefault());
+		}
+
+		/// <summary>
+		///		Starts the playback of the given <see cref="VideoItem"/>.
+		/// </summary>
+		/// <param name="value"></param>
+		public async Task Play(VideoItem value)
+		{
+			await BlazoredVideo.PausePlayback();
+			_playAfterRender = true;
+			CurrentItem = value;
+			await OnNextPlayed.InvokeAsync();
+			StateHasChanged();
+		}
+
+		/// <summary>
+		///		Adds a new item into the queue at its end and updates the <see cref="CurrentItem"/> if necessary.
+		/// </summary>
+		/// <param name="videoItem"></param>
+		public void AddVideoItem(VideoItem videoItem)
+		{
+			if (VideoItems.Any(f => f.Id == videoItem.Id))
+			{
+				return;
+			}
+			VideoItems.Add(videoItem);
+			if (VideoItems.Count == 1)
+			{
+				CurrentItem = videoItem;
+			}
+			StateHasChanged();
+		}
+
+		private async ValueTask TryPlayItem(VideoItem next)
+		{
+			await BlazoredVideo.PausePlayback();
+			if (Repeat is RepeatValues.Loop && next is null)
+			{
+				_playAfterRender = true;
+				CurrentItem = VideoItems.FirstOrDefault();
+				StateHasChanged();
+				await OnNextPlayed.InvokeAsync();
+			}
+			else if (next is not null)
+			{
+				_playAfterRender = true;
+				CurrentItem = next;
+				StateHasChanged();
+				await OnNextPlayed.InvokeAsync();
+			}
+			else
+			{
+				await OnPlaylistEnded.InvokeAsync();
+				await BlazoredVideo.PausePlayback();
+			}
+		}
+
+		protected override void OnInitialized()
+		{
+			if (QueueData is { Length: > 0 })
+			{
+				foreach (var queueItem in QueueData)
+				{
+					var item = new VideoItem();
+#pragma warning disable BL0005
+					item.Source = queueItem;
+#pragma warning restore BL0005
+					VideoItems.Add(item);
+				}
+			}
+
+			if (BlazoredVideo.EndedEvent.HasDelegate)
+			{
+				throw new InvalidOperationException("Cannot subscribe to 'BlazoredVideo.EndedEvent' as it is already in use. Please unsubscribe from the event.");
+			}
+
+#pragma warning disable BL0005
+			BlazoredVideo.EndedEvent = new EventCallback<VideoState>(this, OnPlaybackEnded);
+#pragma warning restore BL0005
+		}
+
+		private async ValueTask OnPlaybackEnded(VideoState videoState)
+		{
+			if (Delay != 0)
+			{
+				await Task.Delay(Delay);
+			}
+
+			if (Repeat is RepeatValues.Once)
+			{
+				await BlazoredVideo.SetCurrentTimeAsync(0);
+				if (await BlazoredVideo.GetPausedAsync())
+				{
+					await BlazoredVideo.StartPlayback();
+				}
+
+				await OnNextPlayed.InvokeAsync();
+			}
```

Comment:
~~`RepeatValues.Once` doesn't seem to work - it repeats the current video continuously, while I would expect it to either repeat each video once (play video 1 twice, then video 2 twice, etc.) OR play the entire queue twice~~

This was my misunderstanding what "Once" meant - so I think that enum name is misleading (see deleted comment above) - could it be "Current" instead of once - as something that repeats "once" is not the same as something that repeats the current video endlessly.

Reply:
I more or less took the names from VLC for its repeat modes. 
![image](https://user-images.githubusercontent.com/6794763/210823974-158b619f-5fea-4c22-a611-84ee7d7b797c.png)